### PR TITLE
Ressurect si114x PR #523

### PR DESCRIPTION
**Pull request by bdilly**
Make all fixes and update license

@edersondisouza ptal

Signed-off-by: Bruno Dilly <bruno.dilly@intel.com>

### CODE REVIEW

File: src/modules/flow/si114x/si114x.c [hunk 1]
```diff
@@ -220,49 +207,62 @@ busy_bus_callback(void *data)
...
     if (status < 0) {
         SOL_WRN("Couldn't open the si114x hardware for usage, please check the pinage.");
         return;
     }
 
-    data = data; /* silence warning */
-
-    if (mdata->init_step >= ARRAY_SIZE(initialization_data)) {
+    if (mdata->init_step >= SOL_UTIL_ARRAY_SIZE(initialization_data)) {
         while (mdata->pendent_calls) {
```
Comment (edersondisouza) [inline src/modules/flow/si114x/si114x.c]: 'pendent' sounds strange... Isn't 'pending' better?

File: src/modules/flow/si114x/si114x.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            "description": "Sensor reading speed",
+            "name": "speed"
+          }
+        ],
+        "version": 1
+      },
+      "in_ports": [
+        {
+          "data_type": "any",
+          "description": "Signal the s2p chip that a read must be performed",
```
Comment (edersondisouza) [inline src/modules/flow/si114x/si114x.json]: 's2p'?

File: src/modules/flow/si114x/si114x.c [hunk 3]
```diff
@@ -304,14 +312,20 @@ do_processing(void *data)
...
     struct si114x_data *mdata = data;
 
-    if (!sol_i2c_busy(mdata->context)) {
-        if (!sol_i2c_read_register(mdata->context, REG_AUX_UVINDEX0, (uint8_t *)&mdata->read_data, 2, &read_callback, mdata)) {
-            sol_flow_send_error_packet(mdata->node, EIO, "Couldn't read from device, check your UV reader (si114x)");
-            SOL_WRN("Couldn't read from device, check your UV reader (si114x)");
-        }
-    } else {
+    if (sol_i2c_busy(mdata->context)) {
         sol_timeout_add(0, &do_processing, mdata);
```
Comment (edersondisouza) [inline src/modules/flow/si114x/si114x.c]: Shouldn't we care about killing this timer?

**Comment by bdilly** [comment]
@edersondisouza I fixed all issues you've spotted in this last push.
Just waiting for CI results and I'm going to merge it.
Thanks

**Comment by bdilly** [comment]
done, thanks

**Pull request closed without merging**